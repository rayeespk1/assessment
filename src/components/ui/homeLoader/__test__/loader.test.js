import React from 'react';
import '@testing-library/jest-dom';
import { render, cleanup } from "@testing-library/react";
import Loading from '../loader';

afterEach(cleanup);

test("renders", () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
});

test("loading correct data", () => {
    const { getByTestId, getByText } = render(<Loading />);
    expect(getByTestId("loadingTag")).toHaveTextContent("Loading...");
    expect(getByTestId("logoTag")).toHaveClass("zoom-ani");
});