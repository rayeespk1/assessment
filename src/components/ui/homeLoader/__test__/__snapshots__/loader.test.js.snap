// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    id="loading"
  >
    <div
      class="one"
    >
      <div
        class="two"
      />
    </div>
    <div
      class="three"
    >
      <div
        class="four"
      />
      <div
        class="five"
      >
        <img
          alt="Rak Bank"
          class="zoom-ani"
          data-testid="logoTag"
          src="logo.png"
          width="130px"
        />
      </div>
      <div
        id="nlpt"
      />
      <div
        class="six msg"
        data-testid="loadingTag"
      >
        Loading...
      </div>
    </div>
  </div>
</DocumentFragment>
`;
